torch/nn/gradients.py

```diff
@@ -0,0 +1,209 @@
+"""Gradient interface"""
+
+import torch
+from .modules.utils import _pair, _triple
+
+
+def _output_padding(input, output_size, stride, padding, kernel_size):
+    output_size = list(output_size)
+    k = input.dim() - 2
+    print(k)
+    print(output_size)
+    if len(output_size) == k + 2:
+        output_size = output_size[-k:]
+    if len(output_size) != k:
+        raise ValueError("output_size must have {} or {} elements (got {})"
+                         .format(k, k + 2, len(output_size)))
+
+    def dim_size(d):
+        return ((input.size(d + 2) - 1) * stride[d] - 2 * padding[d] +
+                kernel_size[d])
+
+    min_sizes = [dim_size(d) for d in range(k)]
+    max_sizes = [min_sizes[d] + stride[d] - 1 for d in range(k)]
+    for size, min_size, max_size in zip(output_size, min_sizes, max_sizes):
+        if size < min_size or size > max_size:
+            raise ValueError(
+                ("requested an output size of {}, but valid sizes range "
+                 "from {} to {} (for an input of {})").format(
+                     output_size, min_sizes, max_sizes,
+                     input.size()[2:]))
+
+    return tuple([output_size[d] - min_sizes[d] for d in range(k)])
+
+
+def conv2d_input(in_channels, out_channels, out_backprop, weight, input_size,
```

Comment:
I like the other convention better, where you have `input_size, weight`. The idea is that the relative ordering of input and weight is always the same.

Reply:
This would also be consistent with TF's argument ordering https://www.tensorflow.org/versions/r1.0/api_docs/python/tf/nn/conv2d_backprop_input